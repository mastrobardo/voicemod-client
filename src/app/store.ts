import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import { soundsApi } from '../features/MainView/main-view.service';
import soundDetailsSliceReducer from '../features/SoundDetails/sound-details.slice';
import logger from 'redux-logger';

export const store = configureStore({
  reducer: {
    [soundsApi.reducerPath]: soundsApi.reducer,
    soundsDetails: soundDetailsSliceReducer
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat([soundsApi.middleware, logger]),
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
