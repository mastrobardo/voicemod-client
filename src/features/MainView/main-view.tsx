import React, { useEffect } from "react";
import { soundsApi } from "./main-view.service";
import { Sound } from "./main-view.types";
import { Box, Grid } from "@mui/material";
import { MediaCard } from "../Card/Card";
import { NoSound } from "../NoSounds/no-sounds";
import { SoundDetails } from "../SoundDetails/sound-details";
import { useAppSelector } from "../../app/hooks";

export const MainView: React.FC = () => {
    const result = soundsApi.endpoints.getSounds.useQuery();
    const hasItems = result && result.data?.length;
    const modalId = useAppSelector((state: any) => state.soundsDetails.modalId);

    useEffect(() => {
        console.log(modalId)
    }, [modalId])

    if(!hasItems) {
        return <NoSound />
    }

    return (
        <Box sx={{ p: 2 }} >
        <SoundDetails id={modalId} open={Boolean(modalId)} />
        <Grid container spacing={3} alignItems="center" justifyContent="center">    
            {hasItems && result?.data?.map((ele:Sound, idx: number) => {
                return (
                    <Grid key={ele._id} item xs={12} sm={6} lg={3}>
                        <MediaCard {...ele} />
                    </Grid>
                )
            })}  
        </Grid>
        </Box>
    )
}