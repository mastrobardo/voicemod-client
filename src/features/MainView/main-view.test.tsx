import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { store } from '../../app/store';

import { MainView } from './main-view';
describe('<MediaCard>', () => {
    const initialState = {}    

    it('renders correctly when there are no items', () => {
        const tree = renderer.create(<Provider store={store}><MainView /></Provider>).toJSON();
        expect(tree).toMatchSnapshot();
      });
})
