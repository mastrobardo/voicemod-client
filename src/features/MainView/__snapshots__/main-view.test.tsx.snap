// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MediaCard> renders correctly when there are no items 1`] = `
<div
  className="MuiBox-root css-hpgf8j"
>
  <div
    className="MuiTypography-root MuiTypography-h1 MuiTypography-gutterBottom css-19bbok8-MuiTypography-root"
  >
    So empty! You should add some sounds to the database!
  </div>
</div>
`;
